custom/src/main/java/co/elastic/otel/dynamicconfig/CentralConfig.java

```diff
@@ -0,0 +1,271 @@
+      return;
+    }
+    String serviceName = getServiceName(properties);
+    // TODO agree on polling interval property name
+    int pollingInterval = properties.getInt("elastic.otel.opamp.polling.interval_in_seconds", 30);
+    // TODO derive default endpoint from main endpoint when EDOT collector endpoint is stable
+    String endpoint = properties.getString("elastic.otel.opamp.endpoint", "http://localhost:4320");
+    if (!endpoint.endsWith("v1/opamp")) {
+      if (endpoint.endsWith("/")) {
+        endpoint += "v1/opamp";
+      } else {
+        endpoint += "/v1/opamp";
+      }
+    }
+    logger.info("============= Starting OpAmp client for: " + serviceName);
+    DynamicInstrumentation.setTracerConfigurator(
+        providerBuilder, DynamicConfiguration.UpdatableConfigurator.INSTANCE);
+    CentralConfigurationManager centralConfigurationManager =
+        CentralConfigurationManager.builder()
+            .setServiceName(serviceName)
+            .setPollingInterval(Duration.ofSeconds(pollingInterval))
+            .setConfigurationEndpoint(endpoint)
+            .build();
+
+    centralConfigurationManager.start(
+        configuration -> {
+          logger.fine("Received configuration: " + configuration);
+          Configs.applyConfigurations(configuration);
+          return CentralConfigurationProcessor.Result.SUCCESS;
+        });
+
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () -> {
+                  logger.info("=========== Shutting down OpAmp client for: " + serviceName);
+                  centralConfigurationManager.stop();
+                }));
+  }
+
+  private static String getServiceName(ConfigProperties properties) {
+    String serviceName = properties.getString("otel.service.name");
+    if (serviceName != null) {
+      return serviceName;
+    }
+    Map<String, String> resourceMap = properties.getMap("otel.resource.attributes");
+    if (resourceMap != null) {
+      serviceName = resourceMap.get("service.name");
+      if (serviceName != null) {
+        return serviceName;
+      }
+    }
+    return "unknown_service"; // Specified default
+  }
+
+  public static class Configs {
+    private static final Map<String, ConfigOption> configNameToConfig = new HashMap<>();
+    private static final Set<String> currentNonDefaultConfigsApplied = new HashSet<>();
+
+    static {
+      ConfigOption option;
+      configNameToConfig.put((option = new SendLogs()).getConfigName(), option);
+      configNameToConfig.put((option = new SendMetrics()).getConfigName(), option);
+      configNameToConfig.put((option = new SendTraces()).getConfigName(), option);
+      configNameToConfig.put(
+          (option = new DeactivateAllInstrumentations()).getConfigName(), option);
+      configNameToConfig.put((option = new DeactivateInstrumentations()).getConfigName(), option);
+      configNameToConfig.put((option = new LoggingLevel()).getConfigName(), option);
```

Comment:
Nit: A matter of tase but I prefer the functional approach here:
```
configNameToConfig = Stream.of(
    new SendLogs(),
    new SendMetrics(),
    new SendTraces(),
    new DeactivateAllInstrumentations(),
    new DeactivateInstrumentations(),
    new LoggingLevel()
).collect(Collectors.toMap(ConfigOption::getConfigName, option -> option))
```

Reply:
that is nicer, I agree